### DISCUSSION TITLE
Problem with larger supercell phonon power spectrum and effect of host sublattice on ion diffusion

### DISCUSSION THREAD

**Discussion by ballal09** [deepmodeling/deepmd-kit#1660] [Q&A]
Hello

I hope you are doing well.  I am studying an ion conductor and trying to build a DeepMD potential. I have generated my training data by performing AIMD simulation at different temperatures using a 1 1 4 supercell (120 atoms).  I have used se_e2_a model to train my data. My RMSE of force is around 50 meV/A and  for energy it is around  2 meV/atom which is very reasonable. After generating the DeepMD potential, I have validated it against the AIMD data using pair distribution function, mean squared displacement of the diffusing species and phonon power spectrum at different temperature. All of the comparison provided very satisfactory result when I compared AIMD result with the 1 1 4 supercell (120 atoms) model of DeepMD potential.  At this point, I am facing two problems which are discussed below:

**Problem 1: Divergence of larger supercell phonon power spectrum**
I am interested in calculating phonon power spectrum at larger supercell. So, When I increased the supercell to 2 2 4 (480 atoms) in the lammps simulation, my calculated power spectrum is giving incorrect values. I have attached the plots below for your reference. Here, MLMD in the figure stands for machined learned potential obtained from DeepMD.

![phononspectrum_T200K](https://user-images.githubusercontent.com/60304718/165191985-c406c4bd-cca8-476a-a468-f269e11bd4b8.png)
                                      **Fig1:** 1 1 4 supercell power spectrum is in a good match with AIMD power spectrum

After increasing the supercell to 2 2 4, the power spectrum in the high frequency range looks spurious as shown in the plot below.

![FPS_MLMD_T200K](https://user-images.githubusercontent.com/60304718/165193173-4707c4c2-f708-4a48-b2bd-4f3b1c888106.png)
 **Fig 2:** Spurious phonon power spectrum for 2 2 4 supercell at the high frequency regime

Please let me know your thoughts on this suspicious behavior of phonon power spectrum for larger supercell. Thanks in advance!

**Problem 2 : Incorrect prediction of effect of host sublattice on ion diffusion**

When I have frozen the host sublattice in my AIMD simulation, I have found that my diffusing species almost stop diffusing. But, in case of the DeepMD generated potential, When I have frozen the host sublattice, I have found the opposite trend which is unphysical.
For this problem, It has nothing to do with the supercell size. 

For your information, I have attached my input file for training the model.

**Input file:**

{
    "_comment": " model parameters",
    "model": {
	"type_map":	["species 1", "species 2", "species 3"],
	"descriptor" :{
	    "type":		"se_e2_a",
	    "sel":		[50, 20, 70],
	    "rcut_smth":	0.50,
	    "rcut":		6.00,
	    "neuron":		[25, 50, 100],
	    "resnet_dt":	false,
	    "axis_neuron":	16,
	    "seed":		1,
	    "_comment":		" that's all"
	},
	"fitting_net" : {
	    "neuron":		[240, 240, 240],
	    "resnet_dt":	true,
	    "seed":		1,
	    "_comment":		" that's all"
	},
	"_comment":	" that's all"
    },

    "learning_rate" :{
	"type":		"exp",
	"decay_steps":	5000,
	"start_lr":	0.001,	
	"stop_lr":	3.51e-8,
	"_comment":	"that's all"
    },

    "loss" :{
	"type":		"ener",
	"start_pref_e":	0.02,
	"limit_pref_e":	1,
	"start_pref_f":	1000,
	"limit_pref_f":	100,
	"start_pref_v":	0,
	"limit_pref_v":	0,
	"_comment":	" that's all"
    },

    "training" : {
	"training_data": {
	    "systems":		["./data/data_0/"],
	    "batch_size":	"auto",
	    "_comment":		"that's all"
	},
	"validation_data":{
	    "systems":		["./data/data_1/"],
	    "batch_size":	"auto",
	    "numb_btch":	1,
	    "_comment":		"that's all"
	},
	"numb_steps":	200000,
	"seed":		10,
	"disp_file":	"lcurve.out",
	"disp_freq":	100,
	"save_freq":	1000,
	"_comment":	"that's all"
    },    

    "_comment":		"that's all"
}

I appreciate your valuable feedback. Thank you very much for your time!

Best
Ballal

**Comment by wanghan-iapcm** [comment]
Please report the version of deepmd-kit you are using. 

Is there any reason for setting limit_pref_f to 100? In this case it seems that the energy prefactor is much lower than that of the force, which may lead to a low prediction accuracy in energy. 

What is the training and test (if available) accuracy of your model?  

> When I have frozen the host sublattice in my AIMD simulation, I have found that my diffusing species almost stop diffusing. But, in case of the DeepMD generated potential, When I have frozen the host sublattice, I have found the opposite trend which is unphysical.
> For this problem, It has nothing to do with the supercell size.

You may compare the DFT energy and force with the DP energy and force in the same configuration, and see the error of your DP model.

**Comment by ballal09** [comment]
Dear Dr. Han Wang

Thank you very much for your response. 

**Version of DeepMD-kit:**   v2.0.3

**Is there any reason for setting limit_pref_f to 100:**   I was trying to reduce the force error as I already had very low error in energy.

**What is the training and test (if available) accuracy of your model? :** 
**Test Accuracy:  Energy RMSE        : 1.291795e-01 eV,  Force  RMSE        : 4.604140e-02 eV/A**

**Problem 2: Incorrect prediction of effect of host sublattice on ion diffusion**
I was able to fix the problem of effect of host sublattice   on ion diffusion. It was actually a Lammps issue regarding how to  frezze the host sublattice. Now, DeepMD potential correctly reproduce the trend observed in AIMD.

**Problem 1: Divergence of larger supercell phonon power spectrum**

As I mention previously, I have trained my model using AIMD data from 1 1 4 supercell (120 atoms) and found excellent match between AIMD and DeepMD (of course 1 1 4 supercell)  when compared pair distribution function, mean squared displacement and phonon power spectra across different temperature.  But, When I compared the DFT energy & forces to DeepMD energy & forces for the corresponding structures, it was not a perfect y=x line. There was some anomalous behavior although match may be okay (let me know your thought on this). I have attached the plot below for the 1 1 4 supercell.  Please let me know your valuable feedback on how to further improve the correspondence between the DFT and DeepMD computed energy & forces.

![image](https://user-images.githubusercontent.com/60304718/175199139-03a46009-635d-4824-8e69-0680d1e2952d.png)

![image](https://user-images.githubusercontent.com/60304718/175199231-670e4123-8a9d-4abb-b004-44d96c5ba29b.png)

As mentioned previously in the Problem 1, when I increased the supercell size  in Lammps simulation, I have found the incorrect phonon power spectrum in the higher frequency regime. When I compared (for the larger supercell) the energies & forces from DFT and DeepMD for the corresponding structures, I have found the deviation is much larger. Probably this larger deviation is causing anomalous behavior in phonon power spectrum.   I have attached the plot below for 2 2 3 supercell. I highly appreciate your valuable feedback on this. 

![image](https://user-images.githubusercontent.com/60304718/175202402-230e0b15-4c15-47d8-871c-2c64b8c57a73.png)

![image](https://user-images.githubusercontent.com/60304718/175202472-a0e8ca2d-b709-45e6-8743-b2459805a457.png)


**Note on Energy and Force calculation:**  I have grabbed 100 structures from the machined learned molecular dynamics simulation and then just run one ionic loop using VASP to compute the DFT energy and forces. The magnitude of the forces is averaged over all the atoms in the supercell. 


Thank you very much  for your time.

Best
Ballal Ahammed

**Comment by wanghan-iapcm** [comment]
Reading from you plot, the energy error in the 114 supercell is roughly 2meV/atom. Considering that you are setting limit_pref_f to 100 times larger than limit_pref_e, I would not say the error is large. Note that the variance of the energy in your data is only 13meV/atom, so the diversity of your data is very low. That's why you observe a gap between dft-dp energy and  y=x. 

The range of the energy in the plot of 223 supercell is even smaller, only 5meV/atom, so an energy error of 2meV/atom is obvious. 

I do not understand the force error plots. Neither the dft nor the dp force is centered on 0. Please check if you have processed the force labels correctly.

**Comment by namtranlab** [comment]
Could you plot the distribution of your energy and force?
